receiver/googlecloudmonitoringreceiver/receiver_test.go

```diff
@@ -0,0 +1,872 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package googlecloudmonitoringreceiver
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"testing"
+	"time"
+
+	monitoring "cloud.google.com/go/monitoring/apiv3/v2"
+	"cloud.google.com/go/monitoring/apiv3/v2/monitoringpb"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/receiver/scraperhelper"
+	"go.uber.org/zap"
+	"google.golang.org/api/option"
+	"google.golang.org/genproto/googleapis/api/distribution"
+	"google.golang.org/genproto/googleapis/api/metric"
+	"google.golang.org/genproto/googleapis/api/monitoredres"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/durationpb"
+	"google.golang.org/protobuf/types/known/structpb"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type fakeMetricServiceServer struct {
+	monitoringpb.UnimplementedMetricServiceServer
+	metricDescriptors []*metric.MetricDescriptor
+	timeSeries        []*monitoringpb.TimeSeries
+}
+
+type MetricsTestServer struct {
+	lis      net.Listener
+	srv      *grpc.Server
+	Endpoint string
+}
+
+func NewFakeMetricTestServer() (*MetricsTestServer, error) {
+	srv := grpc.NewServer()
+	lis, err := net.Listen("tcp", "localhost:0")
+	if err != nil {
+		return nil, err
+	}
+	testServer := &MetricsTestServer{
+		Endpoint: lis.Addr().String(),
+		lis:      lis,
+		srv:      srv,
+	}
+
+	monitoringpb.RegisterMetricServiceServer(
+		srv,
+		&fakeMetricServiceServer{
+			metricDescriptors: mockMetricDescriptors(),
+			timeSeries:        mockTimeSeriesData(),
+		},
+	)
+
+	return testServer, nil
+}
+
+func mockMetricDescriptors() []*metric.MetricDescriptor {
+	return []*metric.MetricDescriptor{
+		{
+			Name: "projects/fake-project/metricDescriptors/metric.type",
+			Type: "custom.googleapis.com/fake_metric",
+		},
+	}
+}
+
+func mockTimeSeriesData() []*monitoringpb.TimeSeries {
+	return []*monitoringpb.TimeSeries{
+		{
+			Metric: &metric.Metric{
+				Type: "custom.googleapis.com/fake_metric",
+			},
+			Resource: &monitoredres.MonitoredResource{
+				Type:   "global",
+				Labels: map[string]string{"project_id": "fake-project"},
+			},
+			Points: []*monitoringpb.Point{
+				{
+					Interval: &monitoringpb.TimeInterval{
+						StartTime: timestamppb.Now(),
+						EndTime:   timestamppb.Now(),
+					},
+					Value: &monitoringpb.TypedValue{
+						Value: &monitoringpb.TypedValue_Int64Value{Int64Value: 42},
+					},
+				},
+			},
+		},
+	}
+}
+
+// Test cases for `initializeClient`
+func TestInitializeClient_Success(t *testing.T) {
+	ctx := context.Background()
+
+	// Set up fake credentials for testing
+	os.Setenv("GOOGLE_APPLICATION_CREDENTIALS", "./../googlecloudspannerreceiver/testdata/serviceAccount.json")
```

Comment:
lets copy the serviceAccount into this component (in a testdata directory) to avoid cross-module interactions.